src/templates/template-prop-types.js

```diff
@@ -0,0 +1,10 @@
+import PropTypes from 'prop-types';
+// Temporary: These are to be replaced with TypeScript types or removed as soon as possible
+
+export const templatePropTypes = {
+  location: PropTypes.shape({
+    search: PropTypes.string,
+  }),
+  pageContext: PropTypes.object,
+  data: PropTypes.object,
+};
```

Comment:
I'm a bit confused why we are adding `PropTypes` while using TypeScript? Unless I'm missing something, everything touched should now be migrated to TS.

Reply:
Feeling time pressure on this one, these are the existing PropTypes that we already have and I'm not sure if I can add TypeScript types for the innermost components/templates and have time to do everything else